### PR TITLE
Add @ConditionalOnProperty to GcpContextAutoConfiguration

### PR DESCRIPTION
**Pull request by ssoydan-strands**
Add conditional property to GCP core auto configuration.
"spring.cloud.gcp.core.enabled" property enables/disables the GCP core auto configuration.

Related document is updated. Unit tests are added.
Implemented for #2147

### CODE REVIEW

**Comment by ssoydan-strands** [comment]
> You changes look good!
> Just one thing -- could you revert unrelated formatting changes introduced by the IDE? They are sensible, but we try to keep formatting PRs separate from functionality PRs.

Done!

File: docs/src/main/asciidoc/core.adoc [hunk 1]
```diff
@@ -24,6 +24,15 @@ dependencies {
 }
 ----
 
+=== Configuration
+
+The following options may be configured with Spring Cloud core.
+
+|===========================================================================
+| Name | Description | Required | Default value
+| `spring.cloud.gcp.core.enabled` | Enables or disables GCP core auto configuration | No | `true`
```
Comment (meltsufin) [inline docs/src/main/asciidoc/core.adoc]: Since we're adding this table, (which we should've always had), we should probably also list `spring.cloud.gcp.credentials.location`, `spring.cloud.gcp.credentials.encoded-key`, `spring.cloud.gcp.project-id`, and `spring.cloud.gcp.credentials.scopes`.

I understand that's not the main focus of this PR. So, I would be OK with it being done in a separate PR.

File: spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/GcpContextAutoConfiguration.java [hunk 2]
```diff
@@ -40,8 +41,10 @@
  * @author Mike Eltsufin
  * @author Elena Felder
  * @author Chengyuan Zhao
+ * @author Serhat Soydan
  */
 @Configuration
+@ConditionalOnProperty(name = "spring.cloud.gcp.core.enabled", havingValue = "true", matchIfMissing = true)
```
Comment (meltsufin) [inline spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/GcpContextAutoConfiguration.java]: We should also add it to `spring-cloud-gcp/spring-cloud-gcp-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json`.

File: docs/src/main/asciidoc/core.adoc (hunk 1 above)
Comment (elefeint) [reply docs/src/main/asciidoc/core.adoc]: I filed a follow-up issue #2151 for this.

Comment (ssoydan-strands) [reply docs/src/main/asciidoc/core.adoc]: Better to do it in a separate PR connected to the new issue #2151 , I guess.
